cpp/daal/src/algorithms/kmeans/oneapi/kmeans_dense_lloyd_batch_kernel_ucapi_impl.i

```diff
@@ -94,186 +107,127 @@ Status KMeansDenseLloydBatchKernelUCAPI<algorithmFPType>::compute(const NumericT
-    auto mindistances              = context.allocate(TypeIds::id<algorithmFPType>(), blockSize, &st);
-    auto candidates                = context.allocate(TypeIds::id<int>(), nClusters, &st);
-    auto candidateDistances        = context.allocate(TypeIds::id<algorithmFPType>(), nClusters, &st);
-    auto partialCandidates         = context.allocate(TypeIds::id<int>(), nClusters * getCandidatePartNum(nClusters), &st);
-    auto partialCandidateDistances = context.allocate(TypeIds::id<algorithmFPType>(), nClusters * getCandidatePartNum(nClusters), &st);
-    auto partialCentroids          = context.allocate(TypeIds::id<algorithmFPType>(), nPartialCentroids * nClusters * nFeatures, &st);
-    auto partialCentroidsCounters  = context.allocate(TypeIds::id<int>(), nPartialCentroids * nClusters, &st);
-    DAAL_CHECK_STATUS_VAR(st);
-
-    auto compute_squares = kernel_factory.getKernel(getComputeSquaresKernelName(nFeatures), &st);
+    auto & context        = Environment::getInstance()->getDefaultExecutionContext();
+    auto & kernel_factory = context.getClKernelFactory();
+    buildProgram(kernel_factory, nClusters, &st);
     DAAL_CHECK_STATUS_VAR(st);
 
-    auto init_distances             = kernel_factory.getKernel("init_distances", &st);
-    auto compute_assignments        = kernel_factory.getKernel("reduce_assignments", &st);
-    auto partial_reduce_centroids   = kernel_factory.getKernel("partial_reduce_centroids", &st);
-    auto merge_reduce_centroids     = kernel_factory.getKernel("merge_reduce_centroids", &st);
-    auto update_objective_function  = kernel_factory.getKernel("update_objective_function", &st);
-    auto compute_partial_candidates = kernel_factory.getKernel("partial_candidates", &st);
-    auto merge_partial_candidates   = kernel_factory.getKernel("merge_candidates", &st);
+    uint32_t blockSize = 0;
+    DAAL_CHECK_STATUS_VAR(getBlockSize(nRows, nClusters, nFeatures, blockSize));
+    DAAL_CHECK_STATUS_VAR(initializeBuffers(nClusters, nFeatures, blockSize));
 
     BlockDescriptor<algorithmFPType> inCentroidsRows;
-    ntInCentroids->getBlockOfRows(0, nClusters, readOnly, inCentroidsRows);
+    DAAL_CHECK_STATUS_VAR(ntInCentroids->getBlockOfRows(0, nClusters, readOnly, inCentroidsRows));
     auto inCentroids = inCentroidsRows.getBuffer();
 
     BlockDescriptor<algorithmFPType> outCentroidsRows;
-    ntOutCentroids->getBlockOfRows(0, nClusters, readWrite, outCentroidsRows);
+    DAAL_CHECK_STATUS_VAR(ntOutCentroids->getBlockOfRows(0, nClusters, readWrite, outCentroidsRows));
     auto outCentroids = outCentroidsRows.getBuffer();
 
     BlockDescriptor<algorithmFPType> objFunctionRows;
-    ntObjFunction->getBlockOfRows(0, nClusters, readWrite, objFunctionRows);
+    DAAL_CHECK_STATUS_VAR(ntObjFunction->getBlockOfRows(0, nClusters, readWrite, objFunctionRows));
     auto objFunction = objFunctionRows.getBuffer();
+    if (inCentroids == NULL || outCentroids == NULL || objFunction == NULL)
```

Comment:
When this can happen?

Reply:
There is  a requirement to check all pointers before access.